File: 0202-happy-number/solution.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        while n !=1:
            if n in seen:
                return False;
            temp =0
            seen.add(n)
            for c in str(n):
                temp = temp+ int(c)**2
            n=temp
        
        if n==1:
            return True
        else:
            return False

        

File: 0202-happy-number/solution.java
class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> set = new HashSet<>();
        while(n!=1){
            if(set.contains(n))return false;
            long temp = 0;
            set.add(n);
            for(char c: (n+"").toCharArray())temp+=Math.pow((c-'0'), 2);
            n=(int)temp;
            // if(n==og) return false;
        }


    return n==1? true: false;    
    }
}
